### PR TITLE
Make native parser and output negotiation imgproxy-compatible

### PR DESCRIPTION
**Pull request by hlindset**
## Summary
- Reworked the native parser around imgproxy-style URL grammar and normalized request fields
- Added imgproxy-compatible output negotiation, including Accept handling, automatic format selection, and cache-key inputs
- Simplified cache key generation and planner behavior for explicit versus automatic output paths
- Expanded unit and property coverage for aliases, parsing edge cases, negotiation invariants, and cache-key stability

## Testing
- `mix format --check-formatted`
- `mix compile --warnings-as-errors`
- `mix test`

## Summary by CodeRabbit

* **New Features**
  * Improved automatic output negotiation (q=0 exclusion, server-preference ordering, clearer Vary/Accept semantics) and explicit output selection via short aliases or @extension.

* **Changes**
  * URL option grammar revamped to declarative resize/size/resizing_type plus gravity/offsets; defaults updated (nullable output format, enlarge/extend flags, gravity offsets).
  * Filesystem cache keys now reflect normalized automatic-output candidates; cache lookup can gracefully skip on key failures.
  * Streaming and encoding made more robust with consistent headers and better error handling.

* **Documentation**
  * README clarified grammar, normalization ordering, negotiation, and cache semantics.

* **Tests**
  * Large test-suite refresh covering parsing, negotiation, caching, planning, processing, streaming, and timeouts.

### CODE REVIEW

File: lib/image_plug/processing_request.ex [hunk 1]
```diff
@@ -1,32 +1,46 @@
...
+          resizing_type: resizing_type(),
+          enlarge: boolean(),
+          extend: boolean(),
+          extend_gravity: gravity_anchor() | nil,
+          extend_x_offset: float() | nil,
+          extend_y_offset: float() | nil,
+          gravity: gravity(),
+          gravity_x_offset: float(),
+          gravity_y_offset: float(),
+          format: output_format() | nil
```
[the remark this answers is not in the document]
Comment (hlindset) [reply lib/image_plug/processing_request.ex:28]: Leaving this unsupported intentionally for this slice.

In imgproxy, `best` is Pro best-format behavior: generate/evaluate candidate encodings and choose the best result. That is materially different from our Accept-header automatic negotiation path, so normalizing `format:best` or `@best` to `format:nil` would give it the wrong semantics.

Current behavior is deliberate:

- parser accepts `format:best`, `f:best`, `ext:best`, and `@best` as valid imgproxy grammar;
- planner rejects `format: :best` before cache/origin access with `{:unsupported_output_format, :best}`;
- later work can implement true best-format support without changing the public grammar.

The design doc also keeps `best` in the later/Pro bucket for that reason.

**Comment by hlindset** [comment]
@coderabbitai full review please

**Comment by hlindset** [comment]
> Tip
> 
> For best results, initiate chat on the files or code changes.
> 
> `@hlindset` Sure! I'll kick off a full review of the PR right away.
> 
> [full_review]
> 
> 🧠 Learnings used

What happened @coderabbitai?

File: lib/image_plug/param_parser/native.ex [hunk 2]
```diff
@@ -71,142 +101,246 @@ defmodule ImagePlug.ParamParser.Native do
...
+      ["", _extension] ->
+        {:error, {:missing_source_identifier, "plain"}}
+
+      [source, ""] ->
+        decode_source_path(source, nil)
+
+      [source, extension] ->
+        with {:ok, format} <- parse_format(extension) do
+          decode_source_path(source, format)
+        end
```
[the remark this answers is not in the document]
Comment (hlindset) [reply lib/image_plug/param_parser/native.ex:129]: This is intentional and matches the approved imgproxy-compatibility design. A trailing raw `@` with an empty suffix is treated as “no plain-source output extension”, not as an extension parse attempt. So `plain/images/cat.jpg@` behaves like `plain/images/cat.jpg` and stays in automatic output selection, and `f:webp/plain/images/cat.jpg@` keeps the explicit `f:webp` instead of clearing it.

Non-empty suffixes are still validated: unknown extensions are parser errors, and `best` is recognized but rejected later as an unsupported planner semantic for this slice. Rejecting dangling `@` here would make this stricter than the compatibility grammar we decided on.

**Comment by hlindset** [comment]
@coderabbitai review please
